From e3fe6eac640e62ddaa1b13a171a0bdd6ed3cb53e Mon Sep 17 00:00:00 2001
From: ejgoldik <ehud.joseph.goldik@realsenseai.com>
Date: Sun, 26 Apr 2026 10:03:11 +0300
Subject: [PATCH] overlay: enable d4xx camera for Orin jp6

Signed-off-by: ejgoldik <ehud.joseph.goldik@realsenseai.com>
---
 overlay/Makefile | 26 ++++++++++++++++++++++++++
 1 file changed, 26 insertions(+)

diff --git a/overlay/Makefile b/overlay/Makefile
index c88bbe2..cdd125f 100644
--- a/overlay/Makefile
+++ b/overlay/Makefile
@@ -60,6 +60,32 @@ dtbo-y += tegra234-p3767-camera-p3768-imx219-A.dtbo
 dtbo-y += tegra234-p3767-camera-p3768-imx219-imx477.dtbo
 dtbo-y += tegra234-p3767-camera-p3768-imx477-C.dtbo
 dtbo-y += tegra234-p3767-camera-p3768-imx477-A.dtbo
+dtbo-y += tegra234-camera-d4xx-overlay.dtbo
+dtbo-y += tegra234-camera-d4xx-overlay-dual.dtbo
+dtbo-y += tegra234-camera-d4xx-overlay-max96712-EVB.dtbo
+dtbo-y += tegra234-camera-d4xx-overlay-max96712-EVB-cams-0-1.dtbo
+dtbo-y += tegra234-camera-d4xx-overlay-fg12-4ch.dtbo
+dtbo-y += tegra234-camera-d4xx-overlay-fg12-4ch-d5xx.dtbo
+dtbo-y += tegra234-camera-d4xx-overlay-fg12-4ch-cams-0-1-2-3.dtbo
+dtbo-y += tegra234-camera-d4xx-overlay-fg12-16ch.dtbo
+dtbo-y += tegra234-camera-d4xx-overlay-fg12-16ch-d5xx.dtbo
+dtbo-y += tegra234-camera-d4xx-overlay-fg12-16ch-PWR-only.dtbo
+dtbo-y += tegra234-camera-d4xx-overlay-fg12-16ch-cams-0-1.dtbo
+dtbo-y += tegra234-camera-d4xx-overlay-fg12-16ch-cams-0-1-d5xx.dtbo
+dtbo-y += tegra234-camera-d4xx-overlay-fg12-16ch-cams-0-1-d5xx-d4xx.dtbo
+dtbo-y += tegra234-camera-d4xx-overlay-fg12-16ch-cams-0-1-2-3.dtbo
+dtbo-y += tegra234-camera-d4xx-overlay-fg12-16ch-cams-0-1-2-3-d5xx-3d4xx.dtbo
+dtbo-y += tegra234-camera-d4xx-overlay-fg12-16ch-cams-0-4.dtbo
+dtbo-y += tegra234-camera-d4xx-overlay-fg12-16ch-cams-0-4-8-12.dtbo
+dtbo-y += tegra234-camera-d4xx-overlay-advantech.dtbo
+dtbo-y += tegra234-camera-d4xx-overlay-avermedia.dtbo
+dtbo-y += tegra234-camera-d4xx-overlay-seeed.dtbo
+dtbo-y += tegra234-camera-d4xx-overlay-seeed-d5xx.dtbo
+dtbo-y += tegra234-camera-d4xx-overlay-seeed-cams-0-1.dtbo
+dtbo-y += tegra234-camera-d4xx-overlay-seeed-cams-0-1-2.dtbo
+dtbo-y += tegra234-camera-d4xx-overlay-seeed-cams-0-1-2-3.dtbo
+dtbo-y += tegra234-camera-d4xx-overlay-d5xx.dtbo
+dtbo-y += tegra234-camera-d4xx-overlay-fg24-4ch-d5xx.dtbo

 ifneq ($(dtb-y),)
 dtb-y := $(addprefix $(makefile-path)/,$(dtb-y))
-- 
2.43.0
