From a1d273ad82e891af30438b99b3aabe6d6af8fc21 Mon Sep 17 00:00:00 2001
From: Alex Gantman <alexander.gantman@intel.com>
Date: Tue, 10 Dec 2019 14:19:31 +0200
Subject: [PATCH] d4xx: fixing RGB format report

Signed-off-by: Alex Gantman <alexander.gantman@intel.com>
---
 drivers/media/i2c/d4xx.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/drivers/media/i2c/d4xx.c b/drivers/media/i2c/d4xx.c
index 990bce548..ebf983c11 100644
--- a/drivers/media/i2c/d4xx.c
+++ b/drivers/media/i2c/d4xx.c
@@ -2143,6 +2143,12 @@ static int ds5_mux_get_fmt(struct v4l2_subdev *sd, struct v4l2_subdev_pad_config
 
 	dev_info(sd->dev, "%s(): %u %p\n", __func__, pad, state->mux.last_set);
 
+	if (state->is_rgb) {
+		tmp.pad = DS5_MUX_PAD_RGB;
+		pad = DS5_MUX_PAD_RGB;
+	} else
+		tmp.pad = 0;
+
 	switch (pad) {
 	case DS5_MUX_PAD_MOTION_T:
 		remote_sd = &state->motion_t.sensor.sd;
@@ -2153,11 +2159,13 @@ static int ds5_mux_get_fmt(struct v4l2_subdev *sd, struct v4l2_subdev_pad_config
 	case DS5_MUX_PAD_EXTERNAL:
 		remote_sd = &state->mux.last_set->sd;
 		break;
+	case DS5_MUX_PAD_RGB:
+		remote_sd = &state->rgb.sensor.sd;
+		break;
 	default:
 		return -EINVAL;
 	}
 
-	tmp.pad = 0;
 
 	/* Locks internally */
 	ret = ds5_sensor_get_fmt(remote_sd, cfg, &tmp);
-- 
2.17.1

