From cebf4aa7ac4b4c6469163b90b99da35b8cea546d Mon Sep 17 00:00:00 2001
From: Emil Jahshan <emil.jahshan@intel.com>
Date: Mon, 10 May 2021 16:07:39 +0300
Subject: [PATCH 09/11] d4xx: set AE default to true

Signed-off-by: Emil Jahshan <emil.jahshan@intel.com>
---
 drivers/media/i2c/d4xx.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/media/i2c/d4xx.c b/drivers/media/i2c/d4xx.c
index 3c6062f05..587bd77cb 100644
--- a/drivers/media/i2c/d4xx.c
+++ b/drivers/media/i2c/d4xx.c
@@ -1521,7 +1521,7 @@ static const struct v4l2_ctrl_config ds5_ctrl_laser_power = {
 	.min = 0,
 	.max = 1,
 	.step = 1,
-	.def = 0,
+	.def = 1,
 };
 
 static const struct v4l2_ctrl_config ds5_ctrl_manual_laser_power = {
@@ -1771,7 +1771,7 @@ static int ds5_ctrl_init(struct ds5 *state)
 		V4L2_EXPOSURE_MANUAL;
 	ctrls->auto_exp = v4l2_ctrl_new_std_menu(hdl, ops,
 						 V4L2_CID_EXPOSURE_AUTO,
-						 max, 0, V4L2_EXPOSURE_AUTO);
+						 max, 0, 1);
 
 	// TODO: this prevents setting of manual exposure
 	// v4l2_ctrl_auto_cluster(3, &ctrls->auto_exp, 0, false);
-- 
2.17.1

