From 0dfadd3325603f106146ad0ed4fbc15ddaba5947 Mon Sep 17 00:00:00 2001
From: Emil Jahshan <emil.jahshan@intel.com>
Date: Thu, 5 Aug 2021 11:03:57 +0300
Subject: [PATCH] add resolutions & remove hardcode

Signed-off-by: Emil Jahshan <emil.jahshan@intel.com>
---
 drivers/media/i2c/d4xx.c | 83 ++++++++++++++++++++++++++--------------
 1 file changed, 54 insertions(+), 29 deletions(-)

diff --git a/drivers/media/i2c/d4xx.c b/drivers/media/i2c/d4xx.c
index 2e4d2e6b1..16f3de7c7 100644
--- a/drivers/media/i2c/d4xx.c
+++ b/drivers/media/i2c/d4xx.c
@@ -431,26 +431,58 @@ static const u8 ds5_framerates[] = {5, 30};
 
 static const u8 ds5_framerate_30 = 30;
 
-static const u8 ds5_framerate_15_25[] = {15, 25};
 static const u8 ds5_framerate_15_30[] = {15, 30};
 
 static const u8 ds5_framerate_25 = 25;
 
+static const u8 ds5_depth_framerate_to_30[] = {5, 15, 30};
 static const u8 ds5_framerate_to_30[] = {5, 10, 15, 30};
 static const u8 ds5_framerate_to_60[] = {5, 15, 30, 60};
 static const u8 ds5_framerate_to_90[] = {5, 15, 30, 60, 90};
+static const u8 ds5_framerate_100[] = {100};
+static const u8 ds5_framerate_90[] = {90};
 
 static const struct ds5_resolution d43x_depth_sizes[] = {
 	{
 		.width = 1280,
 		.height = 720,
-		.framerates = ds5_framerates,
-		.n_framerates = ARRAY_SIZE(ds5_framerates),
+		.framerates = ds5_depth_framerate_to_30,
+		.n_framerates = ARRAY_SIZE(ds5_depth_framerate_to_30),
+	}, {
+		.width =  848,
+		.height = 480,
+		.framerates = ds5_framerate_to_90,
+		.n_framerates = ARRAY_SIZE(ds5_framerate_to_90),
+	}, {
+		.width =  848,
+		.height = 100,
+		.framerates = ds5_framerate_100,
+		.n_framerates = ARRAY_SIZE(ds5_framerate_100),
 	}, {
 		.width =  640,
 		.height = 480,
-		.framerates = ds5_framerates,
-		.n_framerates = ARRAY_SIZE(ds5_framerates),
+		.framerates = ds5_framerate_to_90,
+		.n_framerates = ARRAY_SIZE(ds5_framerate_to_90),
+	}, {
+		.width =  640,
+		.height = 360,
+		.framerates = ds5_framerate_to_90,
+		.n_framerates = ARRAY_SIZE(ds5_framerate_to_90),
+	}, {
+		.width =  480,
+		.height = 270,
+		.framerates = ds5_framerate_to_90,
+		.n_framerates = ARRAY_SIZE(ds5_framerate_to_90),
+	}, {
+		.width =  424,
+		.height = 240,
+		.framerates = ds5_framerate_to_90,
+		.n_framerates = ARRAY_SIZE(ds5_framerate_to_90),
+	}, {
+		.width =  256,
+		.height = 144,
+		.framerates = ds5_framerate_90,
+		.n_framerates = ARRAY_SIZE(ds5_framerate_90),
 	},
 };
 
@@ -484,12 +516,12 @@ static const struct ds5_resolution ds5_sizes[] = {
 
 static const struct ds5_resolution ds5_rlt_rgb_sizes[] = {
     {
-		.width = 1270,
+		.width = 1280,
 		.height = 800,
 		.framerates = ds5_framerate_to_30,
 		.n_framerates = ARRAY_SIZE(ds5_framerate_to_30),
 	}, {
-		.width = 1270,
+		.width = 1280,
 		.height = 720,
 		.framerates = ds5_framerate_to_30,
 		.n_framerates = ARRAY_SIZE(ds5_framerate_to_30),
@@ -523,6 +555,21 @@ static const struct ds5_resolution ds5_rlt_rgb_sizes[] = {
 
 static const struct ds5_resolution ds5_onsemi_rgb_sizes[] = {
 	{
+		.width = 640,
+		.height = 480,
+		.framerates = ds5_framerate_to_90,
+		.n_framerates = ARRAY_SIZE(ds5_framerate_to_90),
+	}, {
+		.width = 960,
+		.height = 720,
+		.framerates = ds5_framerate_to_60,
+		.n_framerates = ARRAY_SIZE(ds5_framerate_to_60),
+	}, {
+		.width = 1280,
+		.height = 720,
+		.framerates = ds5_framerates,
+		.n_framerates = ARRAY_SIZE(ds5_framerates),
+	}, {
 		.width = 1920,
 		.height = 1080,
 		.framerates = ds5_framerates,
@@ -580,27 +627,6 @@ static const struct ds5_format ds5_depth_formats_d43x[] = {
 	},
 };
 
-static const struct ds5_format ds5_depth_formats_d45x[] = {
-	{
-		// TODO: 0x31 is replaced with 0x1e since it caused low FPS in Jetson.
-		.data_type = 0x1e,	/* UYVY */
-		.mbus_code = MEDIA_BUS_FMT_UYVY8_1X16,
-		.n_resolutions = ARRAY_SIZE(d46x_depth_sizes),
-		.resolutions = d46x_depth_sizes,
-	}, {
-		/* First format: default */
-		.data_type = 0x2a,	/* Y8 */
-		.mbus_code = MEDIA_BUS_FMT_Y8_1X8,
-		.n_resolutions = ARRAY_SIZE(d46x_depth_sizes),
-		.resolutions = d46x_depth_sizes,
-	}, {
-		.data_type = 0x24,	/* 24-bit Calibration */
-		.mbus_code = MEDIA_BUS_FMT_RGB888_1X24,	/* FIXME */
-		.n_resolutions = ARRAY_SIZE(d46x_calibration_sizes),
-		.resolutions = d46x_calibration_sizes,
-	},
-};
-
 static const struct ds5_format ds5_depth_formats_d46x[] = {
 	{
 		// TODO: 0x31 is replaced with 0x1e since it caused low FPS in Jetson.
@@ -2558,7 +2584,6 @@ static int ds5_fixed_configuration(struct i2c_client *client, struct ds5 *state)
 		ret = ds5_read(state, DS5_Y_RES_HEIGHT, &yh);
 	if (!ret)
 		ret = ds5_read(state, DS5_DEVICE_TYPE, &dev_type);
-        dev_type = 6;
 	if (ret < 0)
 		return ret;
 
-- 
2.17.1

