From be5f3425e5ebbd20aa09edeb5e174f664856e778 Mon Sep 17 00:00:00 2001
From: Evgeni Raikhel <evgeni.raikhel@intel.com>
Date: Sat, 12 Feb 2022 23:37:19 +0200
Subject: [PATCH] Fix the contributors list

---
 drivers/media/i2c/d4xx.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/media/i2c/d4xx.c b/drivers/media/i2c/d4xx.c
index 96834cec9..dc01c088e 100644
--- a/drivers/media/i2c/d4xx.c
+++ b/drivers/media/i2c/d4xx.c
@@ -3484,7 +3484,10 @@ static struct i2c_driver ds5_i2c_driver = {
 
 module_i2c_driver(ds5_i2c_driver);
 
-MODULE_DESCRIPTION("Intel D4XX camera driver");
-MODULE_AUTHOR("Emil Jahshan (emil.jahshan@intel.com)");
+MODULE_DESCRIPTION("Intel RealSense D4XX Camera Driver");
+MODULE_AUTHOR( "Guennadi Liakhovetski <guennadi.liakhovetski@intel.com>,\n\
+				Nael Masalha <nael.masalha@intel.com>,\n\
+				Alexander Gantman <alexander.gantman@intel.com>,\n\
+				Emil Jahshan <emil.jahshan@intel.com>");
 MODULE_LICENSE("GPL v2");
 MODULE_VERSION("1.0.1.3");
-- 
2.17.1

