From ef05f7f8286d4b27fde2d55269e6bb885663679f Mon Sep 17 00:00:00 2001
From: Shikun Ding <shikun.ding@intel.com>
Date: Wed, 27 Apr 2022 10:02:44 +0800
Subject: [PATCH] Aligned preferred stride on 64-byte boundary

Signed-off-by: Shikun Ding <shikun.ding@intel.com>
---
 include/media/tegra_camera_core.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/media/tegra_camera_core.h b/include/media/tegra_camera_core.h
index 421c22491..1f57648f7 100644
--- a/include/media/tegra_camera_core.h
+++ b/include/media/tegra_camera_core.h
@@ -23,7 +23,7 @@
 /* Width alignment */
 #define TEGRA_WIDTH_ALIGNMENT	1
 /* Stride alignment */
-#define TEGRA_STRIDE_ALIGNMENT	1
+#define TEGRA_STRIDE_ALIGNMENT	64
 /* Height alignment */
 #define TEGRA_HEIGHT_ALIGNMENT	1
 /* Size alignment */
-- 
2.17.1

