From f9dcb337adf7c8f12b6661c5fc81581cf8cdf601 Mon Sep 17 00:00:00 2001
From: Xin Zhang <xin.x.zhang@intel.com>
Date: Thu, 20 Oct 2022 16:10:52 +0800
Subject: [PATCH] Add Image into bindeb-pkg image deb

Signed-off-by: Xin Zhang <xin.x.zhang@intel.com>
---
 scripts/package/builddeb | 1 +
 1 file changed, 1 insertion(+)

diff --git a/scripts/package/builddeb b/scripts/package/builddeb
index 0a2a7372538c..9b84fd617114 100755
--- a/scripts/package/builddeb
+++ b/scripts/package/builddeb
@@ -142,6 +142,7 @@ if [ "$ARCH" = "um" ] ; then
 else
 	cp System.map "$tmpdir/boot/System.map-$version"
 	cp $KCONFIG_CONFIG "$tmpdir/boot/config-$version"
+	cp "arch/$SRCARCH/boot/Image" "$tmpdir/boot/Image-$version"
 fi
 # Not all arches include the boot path in KBUILD_IMAGE
 if [ -e $KBUILD_IMAGE ]; then
-- 
2.25.1

