From 46734ee81e5c6f13856229d06d73e4f2dba6f442 Mon Sep 17 00:00:00 2001
From: Alex Gantman <alexander.gantman@intel.com>
Date: Tue, 29 Oct 2019 20:15:10 +0200
Subject: [PATCH] d4xx: Implementing RGB support

Signed-off-by: Alex Gantman <alexander.gantman@intel.com>
---
 .../tegra186-quill-camera-ds5.dtsi                 | 103 ++++++++++++++++++---
 1 file changed, 89 insertions(+), 14 deletions(-)

diff --git a/kernel-dts/t18x-common-platforms/tegra186-quill-camera-ds5.dtsi b/kernel-dts/t18x-common-platforms/tegra186-quill-camera-ds5.dtsi
index c6229d0..6d31dda 100644
--- a/kernel-dts/t18x-common-platforms/tegra186-quill-camera-ds5.dtsi
+++ b/kernel-dts/t18x-common-platforms/tegra186-quill-camera-ds5.dtsi
@@ -43,7 +43,6 @@
 					compatible = "intel,d4xx";
 					vcc-supply = <&en_vdd_cam>;
 					pwdn-gpios = <&tegra_main_gpio CAM0_PWDN GPIO_ACTIVE_HIGH>;
-					cam-type = "Depth";
 					ports {
 						#address-cells = <1>;
 						#size-cells = <0>;
@@ -66,7 +65,7 @@
 							};
 						};
 					};
-					/* mode0: Y8, mode1: depth D16 */
+					/* mode0: Z16:1280x960, mode1: Z16:960x720 */
 					mode0 {
 						pixel_t = "yuv_uyvy16";
 						num_lanes = "2";
@@ -135,8 +134,42 @@
 						min_framerate = "6";
 						max_framerate = "30";
 					};
-					/*mode2 {
-						pixel_t = "bayer_bggr10";
+				};
+			};
+			i2c@1 {
+				reg = <0x3>; 	// line 18-19 of the I2C switch
+				#address-cells = <0x1>;
+				i2c-mux,deselect-on-exit;
+				#size-cells = <0x0>;
+
+				rgb4m0: rgb4m@10 {
+					status = "ok";
+					reg = <0x10>;
+					compatible = "intel,d4xx_rgbflv";
+					vcc-supply = <&en_vdd_cam>;
+					pwdn-gpios = <&tegra_main_gpio CAM0_PWDN GPIO_ACTIVE_HIGH>;
+					ports {
+						#address-cells = <1>;
+						#size-cells = <0>;
+
+						port@0 {
+							reg = <0>;
+							rgb4m_out0: endpoint {
+								vc-id = <0>;
+								port-index = <3>;
+								bus-width = <2>;
+								remote-endpoint = <&ds5_csi_in1>;
+								src_rgb0: source@0 {
+									data-types = <0x2a
+											0x1e>;
+									virt-chan = <0>;
+								};
+							};
+						};
+					};
+					/* mode0: YUV:1920x1080, mode1: YUV:1280x720 */
+					mode0 {
+						pixel_t = "yuv_uyvy16";
 						num_lanes = "2";
 						active_w = "1920";
 						active_h = "1080";
@@ -146,14 +179,14 @@
 						discontinuous_clk = "yes";
 						dpcm_enable = "false";
 						cil_settletime = "0";
-						line_length = "1920";
+						line_length = "1920"; /* 2200 */
 						mclk_multiplier = "15.625";
 						embedded_metadata_height = "0";
 						min_framerate = "6";
 						max_framerate = "30";
 					};
-					mode3 {
-						pixel_t = "bayer_bggr10";
+					mode1 {
+						pixel_t = "yuv_uyvy16";
 						num_lanes = "2";
 						active_w = "1280";
 						active_h = "720";
@@ -163,15 +196,14 @@
 						discontinuous_clk = "yes";
 						dpcm_enable = "false";
 						cil_settletime = "0";
-						line_length = "1280";
+						line_length = "1280"; /* 2200 */
 						mclk_multiplier = "15.625";
 						embedded_metadata_height = "0";
 						min_framerate = "6";
 						max_framerate = "30";
-					};*/
+					};
 				};
 			};
-
 		};
 	};
 	host1x {
@@ -179,7 +211,7 @@
 			#address-cells = <0x1>;
 			#size-cells = <0x0>;
 
-			num-channels = <1>;
+			num-channels = <2>;
 
 			channel@0 {
 				reg = <0x0>;
@@ -192,7 +224,6 @@
 					port@0 {
 						reg = <0>;
 						status = "ok";
-
 						ds5_csi_in0: endpoint@0 {
 							status = "ok";
 							port-index = <0>;
@@ -213,10 +244,41 @@
 				};
 			};
 
+			channel@1 {
+				reg = <0x1>;
+				status = "ok";
+
+				ports {
+					#address-cells = <0x1>;
+					#size-cells = <0x0>;
+
+					port@0 {
+						reg = <0>;
+						status = "ok";
+						ds5_csi_in1: endpoint@2 {
+							status = "ok";
+							csi-port = <3>;
+							bus-width = <2>;
+							vc-id = <0>;
+							remote-endpoint = <&rgb4m_out0>;
+						};
+					};
+
+					port@1 {
+						reg = <1>;
+						status = "ok";
+						rgb_csi_out1: endpoint@3 {
+							status = "ok";
+							remote-endpoint = <&vi_in1>;
+						};
+					};
+				};
+			};
+
 		};
 
 		vi_base: vi@15700000 {
-			num-channels = <1>;
+			num-channels = <2>;
 
 			ports {
 				#address-cells = <0x1>;
@@ -225,7 +287,6 @@
 				port@0 {
 					reg = <0>;
 					status = "ok";
-
 					vi_in0: endpoint {
 						status = "ok";
 						port-index = <0>;
@@ -233,11 +294,25 @@
 						remote-endpoint = <&csi_out0>;
 					};
 				};
+
+				port@1 {
+					reg = <1>;
+					status = "ok";
+					vi_in1: endpoint {
+						status = "ok";
+						port-index = <3>;
+						bus-width = <2>;
+						remote-endpoint = <&rgb_csi_out1>;
+					};
+				};
 			};
 			sources {
 				source@0 {
 					remote-source = <&src_y0>;
 				};
+				source@1 {
+					remote-source = <&src_rgb0>;
+				};
 			};
 		};
 	};
-- 
2.7.4

