From 44db4958de9813c6cbfa44c8f7053cf9b7bfb084 Mon Sep 17 00:00:00 2001
From: Emil Jahshan <emil.jahshan@intel.com>
Date: Wed, 29 Sep 2021 12:41:20 +0300
Subject: [PATCH] v4l2-ctl incomplete firmware version

Signed-off-by: Emil Jahshan <emil.jahshan@intel.com>
---
 drivers/media/i2c/d4xx.c | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/drivers/media/i2c/d4xx.c b/drivers/media/i2c/d4xx.c
index 08437844d..48876bcc3 100644
--- a/drivers/media/i2c/d4xx.c
+++ b/drivers/media/i2c/d4xx.c
@@ -398,15 +398,20 @@ static int ds5_raw_write(struct ds5 *state, u16 reg, const void *val, size_t val
 static int ds5_read(struct ds5 *state, u16 reg, u16 *val)
 {
 	unsigned int data;
-	int ret = regmap_read(state->regmap, reg, &data);
+	int ret = regmap_raw_read(state->regmap, reg, val, 2);
+	dev_info(&state->client->dev, "%s(): -- i2c read val 0x%04x: 0x%x\n",
+			__func__, reg, *val);
+	ret = regmap_read(state->regmap, reg, &data);
+	dev_info(&state->client->dev, "%s(): i2c read data 0x%04x: 0x%x\n",
+			__func__, reg, data);
 	if (ret < 0)
 		dev_err(&state->client->dev, "%s(): i2c read failed %d, 0x%04x\n",
-			__func__, ret, reg);
+				__func__, ret, reg);
 	else {
-		*val = data & 0xffff;
+		//*val = data & 0xffff;
 		if (state->dfu_dev.dfu_state_flag == DS5_DFU_IDLE)
 			dev_info(&state->client->dev, "%s(): i2c read 0x%04x: 0x%x\n",
-				 __func__, reg, *val);
+					__func__, reg, *val);
 	}
 
 	return ret;
-- 
2.17.1

