From 2fb920400f00b0e8fa5cd5c6ad749b7974740097 Mon Sep 17 00:00:00 2001
From: Nael Masalha <nael.masalha@intel.com>
Date: Wed, 17 Nov 2021 19:53:04 +0200
Subject: [PATCH] Fix start stream reading and handling status register

Signed-off-by: Nael Masalha <nael.masalha@intel.com>
---
 drivers/media/i2c/d4xx.c | 17 +++++++++++------
 1 file changed, 11 insertions(+), 6 deletions(-)

diff --git a/drivers/media/i2c/d4xx.c b/drivers/media/i2c/d4xx.c
index 0a804cc6a..d25b10199 100644
--- a/drivers/media/i2c/d4xx.c
+++ b/drivers/media/i2c/d4xx.c
@@ -98,7 +98,7 @@ enum ds5_mux_pad {
 
 #define DS5_START_POLL_TIME	10
 
-#define DS5_START_MAX_TIME	50
+#define DS5_START_MAX_TIME	300
 
 #define DS5_START_MAX_COUNT	(DS5_START_MAX_TIME / DS5_START_POLL_TIME)
 
@@ -2323,6 +2323,7 @@ static int ds5_mux_s_stream(struct v4l2_subdev *sd, int on)
 	struct ds5 *state = container_of(sd, struct ds5, mux.sd.subdev);
 	u16 streaming, rate, depth_status, rgb_status, s_stream;
 	int ret = 0;
+	unsigned int i = 0;
 
 	dev_info(&state->client->dev, "%s(): %s on = %d\n", __func__, state->mux.last_set->sd.name, on);
 
@@ -2349,8 +2350,6 @@ static int ds5_mux_s_stream(struct v4l2_subdev *sd, int on)
 	}
 
 	if (on) {
-		unsigned int i;
-
 		// TODO: this read seems to cause FW crash, need to debug
 		//ds5_read(state, 0x402, &rate);
 		rate = 0;
@@ -2359,12 +2358,18 @@ static int ds5_mux_s_stream(struct v4l2_subdev *sd, int on)
 			ds5_read(state, 0x1004, &streaming);
 			ds5_read(state, 0x4800, &depth_status);
 			ds5_read(state, 0x4802, &rgb_status);
-			if (/*status || */(streaming & 0xf00) == ((2 << 8)))
+			if ((!state->is_rgb) && ((streaming & 0xFF) == 2))
+				break;
+			if ((state->is_rgb) && ((streaming & 0xF00) == ((2 << 8))))
 				break;
 			msleep_range(DS5_START_POLL_TIME);
 		}
 	}
-	msleep_range(DS5_START_POLL_TIME*50);
+
+	if(on && (DS5_START_MAX_COUNT == i)) {
+		dev_err(&state->client->dev, "%s(): start streaming failed\n", __func__);
+		ret = -EAGAIN;
+	}
 
 	dev_info(&state->client->dev, "%s(): streaming %x depth status 0x%04x, rgb status 0x%04x, rate %u\n", __func__,
 		 streaming, depth_status, rgb_status, rate);
@@ -3241,4 +3246,4 @@ module_i2c_driver(ds5_i2c_driver);
 MODULE_DESCRIPTION("Intel D4XX camera driver");
 MODULE_AUTHOR("Emil Jahshan (emil.jahshan@intel.com)");
 MODULE_LICENSE("GPL v2");
-MODULE_VERSION("1.0.0.3");
+MODULE_VERSION("1.0.0.4");
-- 
2.17.1

