From d4632845b8379cbf1b0ec9fabb8a3da896eae6c7 Mon Sep 17 00:00:00 2001
From: Shikun Ding <shikun.ding@intel.com>
Date: Thu, 21 Apr 2022 13:50:19 +0800
Subject: [PATCH] Bump to new version 1.0.1.7

Signed-off-by: Shikun Ding <shikun.ding@intel.com>
---
 drivers/media/i2c/d4xx.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/media/i2c/d4xx.c b/drivers/media/i2c/d4xx.c
index 3eecd661c..83ff4c034 100644
--- a/drivers/media/i2c/d4xx.c
+++ b/drivers/media/i2c/d4xx.c
@@ -3507,4 +3507,4 @@ MODULE_AUTHOR( "Guennadi Liakhovetski <guennadi.liakhovetski@intel.com>,\n\
 				Qingwu Zhang <qingwu.zhang@intel.com>,\n\
 				Shikun Ding <shikun.ding@intel.com>");
 MODULE_LICENSE("GPL v2");
-MODULE_VERSION("1.0.1.6");
+MODULE_VERSION("1.0.1.7");
-- 
2.17.1

