From d761f7684dbf3809a6d487766b9a5a304f1cc067 Mon Sep 17 00:00:00 2001
From: Xin Zhang <xin.x.zhang@intel.com>
Date: Thu, 20 Oct 2022 16:14:02 +0800
Subject: [PATCH] Fix header_check error in make bindeb-pkg

Signed-off-by: Xin Zhang <xin.x.zhang@intel.com>
---
 include/uapi/net/pgdrv.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/uapi/net/pgdrv.h b/include/uapi/net/pgdrv.h
index 6d9c7562b..a4a36860d 100644
--- a/include/uapi/net/pgdrv.h
+++ b/include/uapi/net/pgdrv.h
@@ -35,7 +35,7 @@
 #ifndef _UAPI_PG_DRV_H
 #define _UAPI_PG_DRV_H
 
-#include <linux/cdev.h>
+//#include <linux/cdev.h>
 
 typedef struct _PCI_CONFIG_RW_ {
 	union {
-- 
2.25.1

