From a3d476386840aa595569f1a348dec299007f7643 Mon Sep 17 00:00:00 2001
From: Xin Zhang <xin.x.zhang@intel.com>
Date: Thu, 8 Dec 2022 21:42:46 +0800
Subject: [PATCH] Add Kconfig VIDEO_D4XX_SERDES

Signed-off-by: Xin Zhang <xin.x.zhang@intel.com>
---
 drivers/media/i2c/Kconfig | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/media/i2c/Kconfig b/drivers/media/i2c/Kconfig
index 90f5a90e6..f51bb49b8 100644
--- a/drivers/media/i2c/Kconfig
+++ b/drivers/media/i2c/Kconfig
@@ -194,6 +194,16 @@ config VIDEO_D4XX
 		To compile this driver as a module, choose M here: the module
 		will be called d4xx.
 
+config VIDEO_D4XX_SERDES
+	bool "D4XX camera sensor setup with Maxim Serdes"
+	depends on VIDEO_D4XX
+        default y
+	help
+		This is a Video4Linux2 sensor-level driver for the Intel
+		D4XX depth camera sensor
+
+                This option is for setup with Maxim Serdes.
+
 endmenu
 
 endif
-- 
2.30.2

