From 02da8355b18c65518c9a181473cbc646cdc84f9f Mon Sep 17 00:00:00 2001
From: ejgoldik <ehud.joseph.goldik@realsenseai.com>
Date: Wed, 25 Mar 2026 09:39:42 +0200
Subject: [PATCH] Add max9296 GPIO tunneling for the external sync support

Signed-off-by: ejgoldik <ehud.joseph.goldik@realsenseai.com>
---
 drivers/media/i2c/max9296.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/drivers/media/i2c/max9296.c b/drivers/media/i2c/max9296.c
index 87f9dcf7a..829657bca 100644
--- a/drivers/media/i2c/max9296.c
+++ b/drivers/media/i2c/max9296.c
@@ -26,6 +26,10 @@
 #include <media/max9296.h>
 
 /* register specifics */
+#define MAX9296_2C5_ADDR 0x2C5
+#define MAX9296_2C6_ADDR 0x2C6
+#define MAX9296_2C7_ADDR 0x2C7
+
 #define MAX9296_DST_CSI_MODE_ADDR 0x330
 #define MAX9296_LANE_MAP1_ADDR 0x333
 #define MAX9296_LANE_MAP2_ADDR 0x334
@@ -54,6 +58,10 @@
 #define MAX9296_CTRL0_ADDR 0x10
 
 /* data defines */
+#define MAX9296_2C5_ESYNC 0xAA
+#define MAX9296_2C6_ESYNC 0x17
+#define MAX9296_2C7_ESYNC 0x47
+
 #define MAX9296_CSI_MODE_4X2 0x1
 #define MAX9296_CSI_MODE_2X4 0x4
 #define MAX9296_LANE_MAP1_4X2 0x44
@@ -364,6 +372,9 @@ int max9296_setup_control(struct device *dev, struct device *s_dev)
 
 	max9296_write_reg(dev,
 			MAX9296_PWDN_PHYS_ADDR, MAX9296_ALLPHYS_NOSTDBY);
+	max9296_write_reg(dev, MAX9296_2C5_ADDR, MAX9296_2C5_ESYNC);
+	max9296_write_reg(dev, MAX9296_2C6_ADDR, MAX9296_2C6_ESYNC);
+	max9296_write_reg(dev, MAX9296_2C7_ADDR, MAX9296_2C7_ESYNC);
 
 	priv->sdev_ref++;
 
-- 
2.43.0

