From 062a6765043e3ec63ca50aa73a0f9e3674a14138 Mon Sep 17 00:00:00 2001
From: ejgoldik <ehud.joseph.goldik@realsenseai.com>
Date: Wed, 25 Mar 2026 14:26:10 +0200
Subject: [PATCH] Allow external address assignment on max9295

Signed-off-by: ejgoldik <ehud.joseph.goldik@realsenseai.com>
---
 drivers/media/i2c/max9295.c | 87 +++++++++++++++++--------------------
 1 file changed, 41 insertions(+), 46 deletions(-)

diff --git a/drivers/media/i2c/max9295.c b/drivers/media/i2c/max9295.c
index 8dd593de7..2ea0ee461 100644
--- a/drivers/media/i2c/max9295.c
+++ b/drivers/media/i2c/max9295.c
@@ -120,6 +120,7 @@ struct max9295 {
 	struct regmap *regmap;
 	struct max9295_client_ctx g_client;
 	struct mutex lock;
+	bool external_addr_reassign;
 	/* primary serializer properties */
 	__u32 def_addr;
 	__u32 pst2_ref;
@@ -323,50 +324,52 @@ int max9295_setup_control(struct device *dev)
 
 	g_ctx = priv->g_client.g_ctx;
 
-	if (prim_priv__) {
-		/* update address reassingment */
-		max9295_write_reg(&prim_priv__->i2c_client->dev,
-				MAX9295_DEV_ADDR, (g_ctx->ser_reg << 1));
-	}
+	if (!priv->external_addr_reassign) {
+		if (prim_priv__) {
+			/* update address reassingment */
+			max9295_write_reg(&prim_priv__->i2c_client->dev,
+					MAX9295_DEV_ADDR, (g_ctx->ser_reg << 1));
+		}
 
-	if (g_ctx->serdes_csi_link == GMSL_SERDES_CSI_LINK_A)
-		err = max9295_write_reg(dev, MAX9295_CTRL0_ADDR, 0x21);
-	else
-		err = max9295_write_reg(dev, MAX9295_CTRL0_ADDR, 0x22);
+		if (g_ctx->serdes_csi_link == GMSL_SERDES_CSI_LINK_A)
+			err = max9295_write_reg(dev, MAX9295_CTRL0_ADDR, 0x21);
+		else
+			err = max9295_write_reg(dev, MAX9295_CTRL0_ADDR, 0x22);
 
-	/* check if serializer device exists */
-	if (err) {
-		dev_err(dev, "%s: ERROR: ser device not found\n", __func__);
-		goto error;
-	}
+		/* check if serializer device exists */
+		if (err) {
+			dev_err(dev, "%s: ERROR: ser device not found\n", __func__);
+			goto error;
+		}
 
-	/* delay to settle link */
-	msleep(100);
+		/* delay to settle link */
+		msleep(100);
 
-	for (i = 0; i < ARRAY_SIZE(addr_offset); i += 3) {
-		if ((g_ctx->ser_reg << 1) == addr_offset[i]) {
-			offset1 = addr_offset[i+1];
-			offset2 = addr_offset[i+2];
-			break;
+		for (i = 0; i < ARRAY_SIZE(addr_offset); i += 3) {
+			if ((g_ctx->ser_reg << 1) == addr_offset[i]) {
+				offset1 = addr_offset[i+1];
+				offset2 = addr_offset[i+2];
+				break;
+			}
 		}
-	}
 
-	if (i == ARRAY_SIZE(addr_offset)) {
-		dev_err(dev, "%s: invalid ser slave address\n", __func__);
-		err = -EINVAL;
-		goto error;
-	}
+		if (i == ARRAY_SIZE(addr_offset)) {
+			dev_err(dev, "%s: invalid ser slave address\n", __func__);
+			err = -EINVAL;
+			goto error;
+		}
 
-	for (i = 0; i < ARRAY_SIZE(i2c_ovrd); i += 2) {
-		/* update address overrides */
-		i2c_ovrd[i+1] += (i < 4) ? offset1 : offset2;
+		for (i = 0; i < ARRAY_SIZE(i2c_ovrd); i += 2) {
+			/* update address overrides */
+			i2c_ovrd[i+1] += (i < 4) ? offset1 : offset2;
 
-		/* i2c passthrough2 must be configured once for all devices */
-		if ((i2c_ovrd[i] == 0x8B) && prim_priv__ &&
-				prim_priv__->pst2_ref)
-			continue;
+			/* i2c passthrough2 must be configured once for all devices */
+			if ((i2c_ovrd[i] == 0x8B) && prim_priv__ &&
+					prim_priv__->pst2_ref)
+				continue;
 
-		max9295_write_reg(dev, i2c_ovrd[i], i2c_ovrd[i+1]);
+			max9295_write_reg(dev, i2c_ovrd[i], i2c_ovrd[i+1]);
+		}
 	}
 
 	/* dev addr pass-through2 ref */
@@ -568,10 +571,6 @@ static int __max9295_set_pipe(struct device *dev, int pipe_id, u8 data_type1,
 {
 	int err = 0;
 	u8 bpp = 0x30;
-	static u8 pipe_x_val = 0x0;
-	struct reg_pair map_multi_pipe_en[] = {
-		{0x0315, 0x80},
-	};
 	struct reg_pair map_bpp8dbl[] = {
 		{0x0312, 0x0F},
 	};
@@ -605,21 +604,17 @@ static int __max9295_set_pipe(struct device *dev, int pipe_id, u8 data_type1,
 
 	map_pipe_control[0].val = 0x40 | data_type1;
 	map_pipe_control[1].val = 0x40 | data_type2;
+	if (pipe_id == 0) {
+		map_pipe_control[1].val |= 0x80;
+	}
 	map_pipe_control[2].val = 1 << vc_id;
 	map_pipe_control[3].val = 0x00;
 	map_pipe_control[4].val = bpp;
 	map_pipe_control[5].val = 0x0E;
 
-	if (pipe_id == 0)
-		pipe_x_val = map_pipe_control[1].val;
-
 	err |= max9295_set_registers(dev, map_pipe_control,
 				     ARRAY_SIZE(map_pipe_control));
 
-	map_multi_pipe_en[0].val = 0x80 | pipe_x_val;
-	err |= max9295_set_registers(dev, map_multi_pipe_en,
-				     ARRAY_SIZE(map_multi_pipe_en));
-
 	return err;
 }
 
-- 
2.43.0

