From 761052d064d68b94a7e09a3f2fbbc9557153489f Mon Sep 17 00:00:00 2001
From: Dmitry Perchanov <dmitry.perchanov@intel.com>
Date: Wed, 18 Oct 2023 14:00:35 +0300
Subject: [PATCH] Porting dts to jetpack 5.1.2

Signed-off-by: Dmitry Perchanov <dmitry.perchanov@intel.com>
---
 tegra194-p2888-0001-p2822-0000.dts | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/tegra194-p2888-0001-p2822-0000.dts b/tegra194-p2888-0001-p2822-0000.dts
index a46befc..7ae66c1 100644
--- a/tegra194-p2888-0001-p2822-0000.dts
+++ b/tegra194-p2888-0001-p2822-0000.dts
@@ -18,7 +18,8 @@
  * Top level DTS file for CVM:P2888-0001 and CVB:P2822-0000.
  */
 #include "common/tegra194-p2888-0001-p2822-0000-common.dtsi"
-#include "common/tegra194-p2822-camera-modules.dtsi"
+#include "common/tegra194-camera-d4xx.dtsi"
+/*#include "common/tegra194-p2822-camera-modules.dtsi"*/
 #include "common/tegra194-audio-p2822-0000.dtsi"
 
 #if defined(LINUX_VERSION) && LINUX_VERSION >= 419
-- 
2.34.1

