Index: kernel-noble-src/arch/arm64/boot/dts/nvidia/Makefile
===================================================================
--- kernel-noble-src.orig/arch/arm64/boot/dts/nvidia/Makefile
+++ kernel-noble-src/arch/arm64/boot/dts/nvidia/Makefile
@@ -30,3 +30,16 @@ dtb-$(CONFIG_ARCH_TEGRA_234_SOC) += tegr
 dtb-$(CONFIG_ARCH_TEGRA_234_SOC) += tegra234-p3740-0002+p3701-0008.dtb
 dtb-$(CONFIG_ARCH_TEGRA_234_SOC) += tegra234-p3768-0000+p3767-0000.dtb
 dtb-$(CONFIG_ARCH_TEGRA_234_SOC) += tegra234-p3768-0000+p3767-0005.dtb
+
+# Fangzhu (FG12-16CH, max96712) overlays for JP7.2 on AGX Orin (tegra234)
+dtb-$(CONFIG_ARCH_TEGRA_234_SOC) += tegra234-camera-d4xx-overlay-fg12-16ch.dtbo
+dtb-$(CONFIG_ARCH_TEGRA_234_SOC) += tegra234-camera-d4xx-overlay-fg12-16ch-cams-0-1-2-3.dtbo
+
+dtb-$(CONFIG_ARCH_TEGRA_264_SOC) += tegra264-camera-d4xx-overlay-advantech.dtbo
+dtb-$(CONFIG_ARCH_TEGRA_264_SOC) += tegra264-camera-d4xx-overlay-advantech-cams-0-1-2-3.dtbo
+dtb-$(CONFIG_ARCH_TEGRA_264_SOC) += tegra264-camera-d4xx-overlay-advantech-cams-4-5.dtbo
+dtb-$(CONFIG_ARCH_TEGRA_264_SOC) += tegra264-camera-d4xx-overlay-advantech-cams-4-5-6-7.dtbo
+dtb-$(CONFIG_ARCH_TEGRA_264_SOC) += tegra264-camera-d4xx-overlay-advantech-cams-0-1-2-3-4-5.dtbo
+dtb-$(CONFIG_ARCH_TEGRA_264_SOC) += tegra264-camera-d4xx-overlay-advantech-cams-0-1-2-3-4-5-6-7.dtbo
+dtb-$(CONFIG_ARCH_TEGRA_264_SOC) += tegra264-camera-d4xx-overlay-max96712-EVB.dtbo
+dtb-$(CONFIG_ARCH_TEGRA_264_SOC) += tegra264-camera-d4xx-overlay.dtbo
