From 91e9bb1de78dbd96dc80d47e1a7c1fd28873e8fa Mon Sep 17 00:00:00 2001
From: Dmitry Perchanov <dmitry.perchanov@intel.com>
Date: Mon, 29 Jan 2024 05:52:20 +0200
Subject: [PATCH] overlay: enable d4xx camera for Orin jp6

Signed-off-by: Dmitry Perchanov <dmitry.perchanov@intel.com>
---
 overlay/Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/overlay/Makefile b/overlay/Makefile
index b3b0638..ff0cb54 100644
--- a/overlay/Makefile
+++ b/overlay/Makefile
@@ -54,6 +54,7 @@ dtbo-y += tegra234-p3737-camera-dual-hawk-ar0234-e3653-overlay.dtbo
 dtbo-y += tegra234-p3737-camera-imx390-overlay.dtbo
 dtbo-y += tegra234-p3737-camera-p3762-a00-overlay.dtbo
 dtbo-y += tegra234-p3740-camera-p3783-a00-overlay.dtbo
+dtbo-y += tegra234-camera-d4xx-overlay.dtbo
 
 
 ifneq ($(dtb-y),)
-- 
2.34.1

