From 00f96bc253e0e35235b221af28bddf25d185d84d Mon Sep 17 00:00:00 2001
From: Dmitry Perchanov <dmitry.perchanov@intel.com>
Date: Mon, 20 May 2024 18:16:07 +0300
Subject: [PATCH] overlay: enable d4xx camera for Orin jp6

Signed-off-by: Dmitry Perchanov <dmitry.perchanov@intel.com>
---
 overlay/Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/overlay/Makefile b/overlay/Makefile
index c88bbe2..3f6354e 100644
--- a/overlay/Makefile
+++ b/overlay/Makefile
@@ -60,6 +60,7 @@ dtbo-y += tegra234-p3767-camera-p3768-imx219-A.dtbo
 dtbo-y += tegra234-p3767-camera-p3768-imx219-imx477.dtbo
 dtbo-y += tegra234-p3767-camera-p3768-imx477-C.dtbo
 dtbo-y += tegra234-p3767-camera-p3768-imx477-A.dtbo
+dtbo-y += tegra234-camera-d4xx-overlay.dtbo
 
 ifneq ($(dtb-y),)
 dtb-y := $(addprefix $(makefile-path)/,$(dtb-y))
-- 
2.34.1

