From d54ee6265684e43c2ce7a1a6f0e2c0b81403d3a4 Mon Sep 17 00:00:00 2001
From: Dmitry Perchanov <dmitry.perchanov@intel.com>
Date: Tue, 10 May 2022 16:17:47 +0300
Subject: [PATCH] DSO-18218: change i2c clock to 100kHz

---
 common/tegra194-camera-d4xx.dtsi | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/common/tegra194-camera-d4xx.dtsi b/common/tegra194-camera-d4xx.dtsi
index 009e0ee..76f760b 100644
--- a/common/tegra194-camera-d4xx.dtsi
+++ b/common/tegra194-camera-d4xx.dtsi
@@ -7,7 +7,7 @@
 / {
 	i2c@3180000 {
 		status = "ok";
-		clock-frequency = <400000>;
+		clock-frequency = <100000>;
 		tca9548@72 {
 			status = "ok";
 			reg = <0x72>;
-- 
2.17.1

