diff --git a/overlay/Makefile b/overlay/Makefile
index 5750473c4..0cf508ea0 100644
--- a/overlay/Makefile
+++ b/overlay/Makefile
@@ -76,6 +76,10 @@ dtbo-y += tegra234-p3767-camera-p3768-imx477-C.dtbo
 dtbo-y += tegra234-p3767-camera-p3768-imx477-A.dtbo
 dtbo-y += tegra234-p3740-0002-p3701-0008-csi.dtbo
 
+# Fangzhu (FG12-16CH, max96712) overlays for JP7.2 on AGX Orin (tegra234)
+dtbo-y += tegra234-camera-d4xx-overlay-fg12-16ch.dtbo
+dtbo-y += tegra234-camera-d4xx-overlay-fg12-16ch-cams-0-1-2-3.dtbo
+
 ifneq ($(dtb-y),)
 dtb-y := $(addprefix $(makefile-path)/,$(dtb-y))
 endif
