From c151da4f104f3f20b603865ffec851fea7b412ad Mon Sep 17 00:00:00 2001
From: Junze Wu <junze.wu@intel.com>
Date: Fri, 10 Jun 2022 10:39:25 +0800
Subject: [PATCH] Disable metadata for IR

Signed-off-by: Junze Wu <junze.wu@intel.com>
---
 common/tegra194-camera-d4xx.dtsi | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/common/tegra194-camera-d4xx.dtsi b/common/tegra194-camera-d4xx.dtsi
index 76f760b..05ddb90 100644
--- a/common/tegra194-camera-d4xx.dtsi
+++ b/common/tegra194-camera-d4xx.dtsi
@@ -204,7 +204,7 @@
 						cil_settletime = "0";
 						line_length = "1280"; /* 2200 */
 						mclk_multiplier = "15.625";
-						embedded_metadata_height = "1";
+						embedded_metadata_height = "0";
 						min_framerate = "5";
 						max_framerate = "30";
 						default_framerate = "30";
-- 
2.25.1

